qmanager/policies/base/queue_policy_base.hpp

```diff
@@ -116,7 +117,7 @@ class queue_policy_base_impl_t
     int process_provisional_cancel ();
     int process_provisional_reprio ();
     int insert_pending_job (std::shared_ptr<job_t> &job, bool into_provisional);
-    int erase_pending_job (std::shared_ptr<job_t> &job, bool &found_in_prov);
+    int erase_pending_job(job_t *job, bool &found_in_prov);
```

Comment:
Nit:
```suggestion
    int erase_pending_job (job_t *job, bool &found_in_prov);
```
Apart from the overhead of `shared_ptr`, is there an advantage to using a raw pointer here?